docs/contributing.rst

```diff
@@ -2,12 +2,701 @@
+repository's code structure. However, there needs to be some manual additions to the
+proper documentation index file(s) for the automation to work.
+
+The index files `docs/user_api/index.rst` and `docs/internal_api/index.rst` (paths
+relative from the root directory) are used for UXarray documentation to allow the
+`User API <user_api/index.rst>`_ and `Internal API <internal_api/index.rst>`_,
+respectively, to be automatically generated.
+
+That being said, the code changes, which might be a new function implementation or some
+modifications to existing ones, must be added to the appropriate `index.rst`
+file so that its documentation page is automatically generated.
+
+.. note::
+    Please also ensure that you are changing the `UXarray draft API
+    <https://github.com/UXARRAY/uxarray/blob/main/docs/user_api/uxarray_api.md>`_
+    accordingly if you are proposing chanes to the API (e.g. new functions/attributes,
+    modification(s) to existing functions, etc.)
+
+3.7.3.5. Usage Examples
+~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note::
+    This may not be required for every PR and can be handled as separate PRs when
+    needed. However, it would be a great practice to provide usage examples in the
+    same PR, especially for demonstrating the use of complex UXarray functions.
+
+The UXarray documentation houses `examples/<example-name>.ipynb` files (paths relative
+from the root directory) to provide `examples.rst>`_ to be automatically generated. If
+you prefer to provide usage examples for the work you have put together, please be sure
+to put your notebook(s) under this same directory.
+
+3.7.4. After You Open The Pull Request
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Here are a few more tips that could be helpful throughout getting your PR merged:
+
+3.7.4.1. Addressing reviews
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now that you have opened a PR that has the necessary elements with it and got reviewers
+assigned, you will hopefully begin to receive reviews soon from them. Please make sure
+to address those review comments, change requests, etc to get your PR ready to be
+merged.
+
+.. note::
+    When you addressed all of a reviewer's comments/requests, do not forget to
+    re-request their next review!
+
+
+3.7.4.2. GitHub Actions checks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+UXarray employs a number of GitHub Actions workflows (please refer to the `GitHub
+Actions <https://docs.github.com/en/actions>`_ guide for detailed information) to make
+sure our PRs, branches, etc. pass certain test scenarios such as the pre-commit hooks,
+code test suite, and documentation generation. The pre-commit hooks workflow ensures
+the code being proposed to be merged is complying with code standards. The test suite
+workflows ensure that the changes are passing for those tests described in `3.7.3.1.
+Unit tests`_ for a matrix of various platforms and Python versions. The documentation
```

Comment:
```suggestion
Unit Tests`_ for a matrix of various platforms and Python versions. The documentation
```
If prior review comment is committed wrt section name

Reply:
That one wasn't committed/ Thanks though!